cmd/gateway-gcs.go

```diff
@@ -459,11 +458,14 @@ func (l *gcsGateway) GetObjectInfo(bucket string, object string) (ObjectInfo, er
 	}
 
 	attrs, err := l.client.Bucket(bucket).Object(object).Attrs(l.ctx)
+
 	if err != nil {
 		return ObjectInfo{}, gcsToObjectError(traceError(err), bucket, object)
 	}
+	objInfo := fromGCSObjectInfo(attrs)
+	objInfo.ETag = fmt.Sprintf("%d", attrs.CRC32C)
```

Comment:
Better make CRC32 as ETag always (irrespective of whether MD5 is set) so that ETag looks "consistent" for all objects. Also there is no particular advantage in making ETag as MD5 when it is available.